src/drt/src/dr/FlexDR_init.cpp

```diff
@@ -1579,68 +1608,43 @@ void FlexDRWorker::initTrackCoords_route(drNet* net,
       auto obj = static_cast<drPathSeg*>(uConnFig);
       const auto [bp, ep] = obj->getPoints();
       const auto lNum = obj->getLayerNum();
+      const auto lNum2 = initTrackCoords_getNonPref(lNum);
+      if (lNum2 < 0) {
+        std::cout << "Error: initTrackCoords cannot add non-pref track"
+                  << std::endl;
+      }
       // vertical
       if (bp.x() == ep.x()) {
         // non pref dir
         if (getTech()->getLayer(lNum)->getDir() == dbTechLayerDir::HORIZONTAL) {
-          if (lNum + 2 <= getTech()->getTopLayerNum()) {
-            xMap[lNum + 2][bp.x()]
-                = nullptr;  // default add track to upper layer
-          } else if (lNum - 2 >= getTech()->getBottomLayerNum()) {
-            xMap[lNum - 2][bp.x()] = nullptr;
-          } else {
-            std::cout << "Error: initTrackCoords cannot add non-pref track"
-                      << std::endl;
-          }
-          // add bp, ep
           yMap[lNum][bp.y()] = nullptr;
           yMap[lNum][ep.y()] = nullptr;
+          if (lNum2 >= 0) {
```

Comment:
If this were an error above then then this check is unnecessary.  Similarly elsewhere

Reply:
Fixed